### PR TITLE
Update product price fixed

### PR DESCRIPTION
**Pull request by essienLeoGreat**

### CODE REVIEW

File: src/electron/dto/update-product-price.input.ts [hunk 1]
```diff
@@ -1,10 +1,11 @@
 export class UpdateProductPriceInput {
 	id: string;
 	
-	// categoryId: string;
+	categoryId: string;
```
Comment (min-proj) [inline src/electron/dto/update-product-price.input.ts:4]: Why do you need categoryId to update product?

Comment (min-proj) [reply src/electron/dto/update-product-price.input.ts:4]: Remember you're only updating the price

File: src/electron/dto/update-product.input.ts [hunk 2]
```diff
@@ -2,21 +2,21 @@ import { Currency } from "../enum/general.enum";
...
-	id: string;
-	
-	categoryId: string;
+  id: string;
 
-	storeId: string;
+  categoryId: string;
 
-	name: string;
+  storeId: string;
```
Comment (min-proj) [inline src/electron/dto/update-product.input.ts:9]: Store is not supposed to be updated. So no need to add it to this payload

File: src/electron/service/product.service.ts [hunk 3]
```diff
@@ -129,18 +129,20 @@ export class ProductService {
...
-    // }
+    const category = await this.categoryService.findOne(updateProductPriceInput.categoryId);
+    if (!category) {
+      throw new Error(`Invalid product category selected : ${updateProductPriceInput.categoryId} `);
+    }
     const newProductChanges = this.productRepository.create(updateProductPriceInput);
 
     let product = { ...existingProduct, ...newProductChanges };
 
+    product.salesPrice = updateProductPriceInput.salesPrice;
```
Comment (min-proj) [inline src/electron/service/product.service.ts:140]: Why do you have to manually set this fields when you have already done that using  `const newProductChanges = this.productRepository.create(updateProductPriceInput);`

File: src/electron/dto/update-product-price.input.ts (hunk 1 above)
Comment (essienLeoGreat) [reply src/electron/dto/update-product-price.input.ts]: I wanted to confirm this from you during our call but it skipped my memory.
Category id was removed from the update-product-price-input and from the update product price service.

**Review by min-proj — APPROVED** [review]

**Pull request merged**